### ISSUE TITLE
Split-user: wconfd tries to chown its listening socket to a different user

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1009]
Originally reported of Google Code with ID 953.

```
In the split user scenario, wconfd runs as gnt-wconfd, however it tries to chown its unix socket to gnt-masterd:

# sudo -u gnt-wconfd strace  ganeti-wconfd -f |& grep ganeti
...
bind(5, {sa_family=AF_FILE, path="/var/run/ganeti/socket/ganeti-wconfd"}, 110) = 0
chown("/var/run/ganeti/socket/ganeti-wconfd", 109, 113) = -1 EPERM (Operation not permitted)

# getent passwd 109
gnt-masterd:x:109:115::/var/lib/ganeti:/bin/false


```
Originally added on 2014-09-23 07:59:47 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The source of this issue is src/Ganeti/WConfd/Server.hs:

serverConfig = ServerConfig GanetiLuxid $ ConnectConfig 60 60

Note that WConfd is setup to use the LuxiD daemon config, instead of its own. This config is passed down to UDSServer.connectServer, which looks up the user for luxid (gnt-masterd). Not sure this is intentional, but it looks like it should be using GanetiWConfd instead.
```
Originally added on 2014-09-23 08:20:47 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Note that config.data is still owned by gnt-masterd, so changing the serverConfig to GanetiWConfd results in wconfd being unable to access config.data.

We either have to change ownership on all masterd-owned files, or have wconfd actually run as the old masterd user.
```
Originally added on 2014-09-23 09:07:51 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 4ddce51deb43bee0d5532b2b9c64d04fe65db235
Author: Apollon Oikonomopoulos <apoikos@gmail.com>
Date:   Tue Sep 23 15:09:06 2014 +0300

    metad: run as root to bind port 80
    
    metad needs to bind port 80, so it should really startup as root. Note
    that this should ideally be fixed, either by changing the port to an
    unprivileged one, or by dropping root privileges after binding the port.
    The latter solution is the preferred one, since using port 80 is
    mandated in the design document.
    
    Signed-off-by: Apollon Oikonomopoulos <apoikos@gmail.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

commit 9decc439eb4a996bc5b7ac8a6852be545a494ecd
Author: Apollon Oikonomopoulos <apoikos@gmail.com>
Date:   Tue Sep 23 15:09:05 2014 +0300

    WConfd: run as <prefix>-masterd by default
    
    Since WConfd is taking over the old masterd role, it is best to have it
    run as the old masterd user. A lot of the files (notably config.data and
    client.pem) are accessible by masterd-only.
    
    This is really just to ease the transition from masterd to wconfd and
    get things working again and should be re-evaluated in the long run.
    
    This fixes issues #952 and #953.
    
    Signed-off-by: Apollon Oikonomopoulos <apoikos@gmail.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

commit abd215fa17a038233d3e1d2533899f8354ad9ce4
Author: Apollon Oikonomopoulos <apoikos@gmail.com>
Date:   Tue Sep 23 15:09:04 2014 +0300

    WConfd: use own server config instead of LuxiD's
    
    Signed-off-by: Apollon Oikonomopoulos <apoikos@gmail.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>


These patches fix the situation for 2.12. The split-user roles for the new
daemons should be reevaluated for the long run. This is tracked in issue 955.
```
Originally added on 2014-09-24 14:48:24 +0000 UTC.

Mentioned Issue with GoogleCode ID 955 has been migrated to GitHub with ID #1011.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-10-10 11:40:54 +0000 UTC.

Changed State: Released
